app/uk/gov/hmrc/transitmovementsauditing/models/Details.scala

```diff
@@ -21,6 +21,8 @@ import play.api.libs.json.Json
 import play.api.libs.json.OFormat
 
 object Details {
-  implicit lazy val detailsFormat: OFormat[Details] = Json.format[Details]
+  implicit lazy val detailsFormat: OFormat[Details]               = Json.format[Details]
+  implicit lazy val detailsRequestFormat: OFormat[DetailsRequest] = Json.format[DetailsRequest]
 }
-case class Details(metadata: Metadata, payload: Option[JsObject])
+case class DetailsRequest(subType: Option[String], metadata: MetadataRequest, payload: Option[JsObject])
+case class Details(details: DetailsRequest)
```

Comment:
No, this is not what I meant. **`Details` should NOT have any knowledge of `DetailsRequest`.** 

My point was that you have a presentation layer model, `DetailsRequest`, which contains **only** what you expect to get from another service. Because this contains a metadata object that is also different between what you get from a service and what you send to auditing, that would also need to be a `MetadataRequest` object which has no knowledge of a `Metadata` object. The ONLY link between `DetailsRequest` and `Details` is some sort of function that maps from `DetailsRequest` to `Details`, and performs the transformations that you need.

You need to properly separate this out.